routers/api/v1/api.go

```diff
@@ -617,7 +617,7 @@ func Routes(ctx gocontext.Context) *web.Route {
 			// setting.CORSConfig.AllowSubdomain // FIXME: the cors middleware needs allowSubdomain option
 			AllowedMethods:   setting.CORSConfig.Methods,
 			AllowCredentials: setting.CORSConfig.AllowCredentials,
-			AllowedHeaders:   []string{"Authorization", "X-Gitea-OTP"},
+			AllowedHeaders:   append(setting.CORSConfig.Headers, "Authorization", "X-Gitea-OTP"),
```

Comment:
```suggestion
			AllowedHeaders:   append([]string{"Authorization", "X-Gitea-OTP"}, setting.CORSConfig.Headers...),
```
I think we have to switch the parameters because `append` may modify the original slice. And that may add these two headers to the web router too.